src/components/form/date-picker/date-picker.tsx

```diff
@@ -0,0 +1,115 @@
+import * as React from "react";
+import {
+  DatePicker,
+  DateView,
+  LocalizationProvider,
+} from "@mui/x-date-pickers";
+import {
+  Controller,
+  ControllerProps,
+  FieldPath,
+  FieldValues,
+} from "react-hook-form";
+import { ForwardedRef, forwardRef } from "react";
+import { AdapterDateFns } from "@mui/x-date-pickers/AdapterDateFns";
+import { getValueByKey } from "@/components/form/time-picker/time-picker";
+import useLanguage from "@/services/i18n/use-language";
+
+type DatePickerFieldProps = {
+  disabled?: boolean;
+  className?: string;
+  views?: readonly DateView[] | undefined;
+  minDate?: Date;
+  maxDate?: Date;
+  autoFocus?: boolean;
+  readOnly?: boolean;
+  label: string;
+  testId?: string;
+  error?: string;
+  defaultValue?: Date | null | undefined;
+};
+
+const DatePickerInput = forwardRef(DatePickerInputRaw) as never as (
+  props: DatePickerFieldProps & {
+    name: string;
+    value: Date | undefined | null;
+    onChange: (value: Date | null) => void;
+    onBlur: () => void;
+  } & { ref?: ForwardedRef<HTMLDivElement | null> }
+) => ReturnType<typeof DatePickerInputRaw>;
+
```

Comment:
```suggestion
type ValueDateType = Date | null | undefined;

type DatePickerFieldProps = {
  disabled?: boolean;
  className?: string;
  views?: readonly DateView[] | undefined;
  minDate?: Date;
  maxDate?: Date;
  autoFocus?: boolean;
  readOnly?: boolean;
  label: string;
  testId?: string;
  error?: string;
  defaultValue?: ValueDateType;
};

const DatePickerInput = forwardRef(DatePickerInputRaw) as never as (
  props: DatePickerFieldProps & {
    name: string;
    value: ValueDateType;
    onChange: (value: ValueDateType) => void;
    onBlur: () => void;
  } & { ref?: ForwardedRef<HTMLDivElement | null> }
) => ReturnType<typeof DatePickerInputRaw>;

```

Also please check all places :)